### PR TITLE
systemctl: use underlines to seperate unit types in listing

### LINKED ISSUE #4137: RFE: in systemctl list-units (and related commands) make unit-type grouping clearer somehow
### Submission type
- [ x ] Request for enhancement (RFE)
### systemd version the issue has been seen with

> 231
### Used distribution

> Ubuntu 16.10 dev

The output of commands like "systemctl list-unit-files" is at default not sorted. But for example having the unit files sorted alphabetically makes it more easy to make a look at them so maybe this could be the default behavior for commands where it is useful.

### PR DESCRIPTION
**Pull request by keszybz**
Fixes #4137.

This should probably be extended to list-unit-files, but I'll let people comment first.

### CODE REVIEW

File: src/systemctl/systemctl.c [hunk 1]
```diff
@@ -488,6 +492,7 @@ static int output_units_list(const UnitInfo *unit_infos, unsigned c) {
                                 printf("%.*s\n", desc_len, "DESCRIPTION");
                         else
                                 printf("%s\n", "DESCRIPTION");
+                        printf("%s", ansi_normal());
```
Comment (poettering) [inline src/systemctl/systemctl.c]: It's probably a good idea to move the `\n` into this `printf()` call as well, as place it _after_ the ANSI sequence, no? At least in "systemd-mount --list" the underline only goes until the rightmost _used_ column and not any further.

File: src/systemctl/systemctl.c [hunk 2]
```diff
@@ -534,6 +546,7 @@ static int output_units_list(const UnitInfo *unit_infos, unsigned c) {
                         printf("%.*s\n", desc_len, u->description);
                 else
                         printf("%s\n", u->description);
+                printf("%s", off_underline);
```
Comment (poettering) [inline src/systemctl/systemctl.c]: similar as above

**Comment by keszybz** [comment]
On Mon, Oct 17, 2016 at 07:06:28AM -0700, Lennart Poettering wrote:

> poettering commented on this pull request.
> 
> Some minor comments...
> 
> Any chance you can attach a screenshot how this looks now?
> 
> > @@ -488,6 +492,7 @@ static int output_units_list(const UnitInfo _unit_infos, unsigned c) {
> >                                  printf("%._s\n", desc_len, "DESCRIPTION");
> >                          else
> >                                  printf("%s\n", "DESCRIPTION");
> > -                        printf("%s", ansi_normal());

I moved it this way on purpose: I was hoping that the terminal emulator
would carry the underline all the way to the right side. I think that
would have looked better. Unfortunately this doesn't happen, at least
with the emulators I tried (gnome-terminal, konsole, urxvt).

I'll move it back.

**Comment by keszybz** [comment]
Updated.

**Pull request merged**